File: exams/maisincho/u2/University/University/University/MongoConnection.py
from pymongo import MongoClient
import urllib.parse

class MongoConnection:
    def __init__(self, user="isaac", isaac="", cluster="cluster0.xaitfht", db_name="universities_db"):
       
        password_escaped = urllib.parse.quote_plus(isaac)
        self.uri = f"mongodb+srv://{user}:{password_escaped}@{cluster}.mongodb.net/?retryWrites=true&w=majority&appName=Cluster0"
        self.db_name = db_name
        self.client = None
        self.db = None

    def connect(self):
        try:
            self.client = MongoClient(self.uri)
            self.db = self.client[self.db_name]
            print(f" Conexión exitosa a MongoDB Atlas: {self.db_name}")
            return self.db
        except Exception as e:
            print(f" Error de conexión a MongoDB: {e}")
            return None

    def close(self):
        if self.client:
            self.client.close()
            print(" Conexión a MongoDB cerrada.")

File: exams/jacome/U2/Exam2_Excercise/Exam2_Excercise/src/main/java/ec/edu/espe/exam2_excercise/controller/MongoConnection.java
package ec.edu.espe.exam2_excercise.controller;
import com.mongodb.client.MongoClient;
import com.mongodb.client.MongoClients;
import com.mongodb.client.MongoDatabase;
/**
 *
 * @author LABS-ESPE
 */
public class MongoConnection {
    private static final String URI = "mongodb+srv://Micaela:Micaela@cluster0.dqlf2xq.mongodb.net/?retryWrites=true&w=majority";
    private static final String DATABASE_NAME = "craftmarket";

    private static MongoClient mongoClient;
    private static MongoDatabase database;
     public static MongoDatabase connect() {
        String uri ="mongodb+srv://machicaiza22:pan@cluster0.dqlf2xq.mongodb.net/?retryWrites=true&w=majority";
        MongoClient client = MongoClients.create(uri);
        return client.getDatabase("craftmarket");
    }
    static {
        try {
            mongoClient = MongoClients.create(URI);
            database = mongoClient.getDatabase(DATABASE_NAME);
            System.out.println("[MongoConnection] Connected to MongoDB Atlas successfully.");
        } catch (Exception e) {
            System.err.println("[MongoConnection] Error connecting to MongoDB: " + e.getMessage());
            database = null;
        }
}
    /**
     * Devuelve la instancia de la base de datos para todo el proyecto.
     * @return MongoDatabase conectado o null si hay error.
     */
    public static MongoDatabase getDatabase() {
        if (database == null) {
            System.err.println("[MongoConnection] Database is null, check your connection.");
        }
        return database;
    }

    /**
     * Cierra la conexión a MongoDB de forma segura.
     */
    public static void closeConnection() {
        if (mongoClient != null) {
            mongoClient.close();
            System.out.println("[MongoConnection] MongoDB connection closed.");
        }
    }
}
